### ISSUE TITLE
[BUG] Wrong cutoff in potential file Au_u3.eam

### ISSUE THREAD

**Issue by t-brink** [lammps/lammps#3209]
**Summary**

In the potential file `Au_u3.eam` that is shipped with LAMMPS, the cutoff value is inconsistent with the range of interatomic distance values: There are 500 data points with dr = 1.1212121212121229e-02, giving a cutoff of 5.6060606. The cutoff in the file is 5.55. That means that parts of the functions are cut off at the tail. I guess the original error comes from the fact that the pair term is zero at 5.55. But the electron density term does seem to be nonzero at 5.55, so a bigger cutoff would be correct in my opinion.

This is probably not noticeable in MD simulations or fcc lattices, but I saw subtle effects in a grain boundary (especially with `dynamical_matrix`), where bond lengths close to the cutoff value can occur. While the error is not that big, the fix is easy, so I think it's worth it. I think this issue is quite evident from the file, but please let me know if you need a demonstration case for this issue, I could try to whip up something simple to reproduce.

The `Au_u3.eam` file is also in the NIST potentials repo and in KIM, so if a fix is deemed necessary, we should also communicate with them.

**LAMMPS Version and Platform**

The file is still the same in current git "develop" branch here on github.

**Comment by akohlmey** [comment]
Sorry, but the fact that the tabulation extends the cutoff is no indication that the cutoff needs to be larger. This may be due to technical reasons. What matters is what the original publication states that documents the potential.
Please also note that the last 5 values of the table are 0.0 and thus you really have only 495 points that matter and "coincidentally" 495*dr = 5.55, i.e. exactly the specified cutoff.

**Comment by t-brink** [comment]
Hmm, you are right about the last five entries even for the electron density, sorry. But, I did not just look at the file and said "oh that's wrong", I actually tested it. Here is for example the pressure component Pxx of a grain boundary calculated by scaling the whole box: https://github.com/t-brink/demo-3209/blob/main/Figure_2.png  You can see that there is a difference between the cutoff in the file (orange) and a cutoff of 5.606... (green). Note that the larger cutoff makes things smoother. The files needed to reproduce this are here: https://github.com/t-brink/demo-3209 (sorry, quick and dirty). I used LAMMPS version "29 Sep 2021 - Update 2".

Could this be due to the interpolation functions being different if there's a trailing zero? Wouldn't it then be then more correct to increase the cutoff, since the specifics of the interpolation are due to LAMMPS, not the original authors (genuine question, not sure what is more "correct", given that the behavior looks weird anyway.) Sorry for not digging deeper initially.

**Comment by akohlmey** [comment]
> since the specifics of the interpolation are due to LAMMPS, not the original authors

Actually, the interpolation and tabulation of the EAM pair styles follow **exactly** the reference implementation of the original EAM code (written in Fortran). This has been carefully tested and is unchanged in LAMMPS for a long time.

Of course it is completely at your own disposal to change the potential, its settings and create a *new* potential file, which would represent an (for your purposes) improved potential. Please note we recently updated on of the tools to generate EAM potential files from original parameters (in the tools/eam_database folder) with improved tabulation at "critical" points. This exists also as a python code for people that are less comfortable with (old style) Fortran. If you look through the databases for potential files, you can see that that would not be the first "improved" potential to be added to those databases.

In short, this is not a bug but rather an unexpected or undesired feature.

**Comment by t-brink** [comment]
OK, makes sense. So I'll have to fiddle with it for my own purposes and specify such changes in any resulting publication. Sorry that that turned out to be more of an "undesired, but correct, feature" than a bug, I was not aware that the fidelity to the original implementation was tested _that_ strictly (but which is of course a good thing).
